### PR TITLE
cmd/lncli: add unused flag to newAddr command

### PR DESCRIPTION
**Pull request by gabbyprecious**
## Change Description
Added unused flag to NewAddr command

## Steps to Test
Steps for reviewers to follow to test the change.

### Testing
- [ ] Your PR passes all CI checks.
- [ ] Tests covering the positive and negative (error paths) are included.
- [ ] Bug fixes contain tests triggering the bug to prevent regressions.

### Code Style and Documentation
- [x] The change obeys the [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation) guidelines, and lines wrap at 80.
- [x] Commits follow the [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure).
- [ ] Any new logging statements use an appropriate subsystem and logging level.
- [x]  [There is a change description in the release notes](https://github.com/lightningnetwork/lnd/tree/master/docs/release-notes), or `[skip ci]` in the commit message for small changes.

📝 Please see our  [Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) for further guidance.

### CODE REVIEW

**Comment by gabbyprecious** [comment]
@guggero and @ellemouton created a new PR, couldn't find a way to resolve the merge commits

**Review by guggero — COMMENTED** [review]
Looks good now, only the PR number in the release notes is incorrect.

You can resolve merge commits by rebasing the git history (ask Google for directions). That cannot be done through the GitHub UI though.

File: docs/release-notes/release-notes-0.15.0.md [hunk 1]
```diff
@@ -27,6 +27,9 @@ then watch it on chain. Taproot script spends are also supported through the
 
 * [Update description for `state` command](https://github.com/lightningnetwork/lnd/pull/6237).
 
+* Add [`--unused`](https://github.com/lightningnetwork/lnd/pull/6375) to
```
[the remark this answers is not in the document]
Comment (gabbyprecious) [reply docs/release-notes/release-notes-0.15.0.md]: changed

**Review by guggero — COMMENTED** [review]
tACK, works as expected. Thanks for adding the feature. Even found a general bug with Taproot addresses while testing this (unrelated to the changes in this PR), see https://github.com/lightningnetwork/lnd/pull/6389.

Needs a rebase, otherwise looks good.

**Review by guggero — APPROVED** [review]

**Comment by gabbyprecious** [comment]
@guggero is there anything still holding on the PR. Would like to pick something else but wanted to be sure this is cleared

**Review by yyforyongyu — APPROVED** [review]
LGTM👍

**Comment by guggero** [comment]
Replaced by/included in https://github.com/lightningnetwork/lnd/pull/7689, original commit author credits are preserved.

**Pull request closed without merging**